### ISSUE TITLE
my training speed is too much slowly

### ISSUE THREAD

**Issue by dahaogewsh** [pytorch/fairseq#2688]
when I training the model just like "transformer_vaswani_wmt_en_de_big" ,and my wps is 1706 ,this is too much slow, I have run the training about one week, and now the training steps is 112000, one epoch has 522288 steps, so I have to run about one month to complete one epoch, this is too slow

my system is :
GPU 1080Ti*3
fairseq 0.9.0
PyTorch 1.2.0
CUDA 10.0
![image](https://user-images.githubusercontent.com/64566598/94981715-08a91080-0567-11eb-8064-a982f2ba7d1f.png)
OS linux
training command is:
fairseq-train $DATADIR \
        --save-dir $CKPTS \
        --dropout 0.2 --relu-dropout 0.2 --attention-dropout 0.2 \
        --tags-data $TAGSDIR \
        --my dropout 0.3 \
        --my chaocan 0.5 \
        --seed 4321 \
        --train-subset train \
        --valid-subset valid \
        --optimizer adam \
        --adam-betas 0.9,0.98 \
        --adam-eps 1e-09 \
        --lr-scheduler inverse_sqrt \
        --lr 0.0007 \
        --min-lr 1e-09 \
        --batch-size 2 \
        --update-freq 64 \
        --save-interval 500000 \
        --max-epoch 26 \
        --no-progress-bar \
        --arch mytransformer_big \
        --task tags_translation \
        --log-interval 1000 \
        --save-interval-updates 1000 \
        --keep-interval-updates 10 \
        --ddp-backend=no_c10d \
        --max-target-positions 512 --max-source-positions 512 \
        --criterion twopeak_cross_entropy \
        --warmup-init-lr 1e-06 \
        --warmup-updates 8000 \
        --mmap \
        --encoder-normalize-before \
        --encoder-layerdrop 0.2 \
        --decoder-layerdrop 0.2

tips:
due to GPU device , I have to restrict my batch-size to 2 and update-freq 64 to make a larger batch-size

**Comment by dahaogewsh** [comment]
![image](https://user-images.githubusercontent.com/64566598/94981939-13fd3b80-0569-11eb-97b3-1652324bd5f6.png)
this is part of my training log

**Comment by dahaogewsh** [comment]
can anyone help me？

**Comment by erip** [comment]
Your batch size is super small. How many params does your model have?

**Comment by dahaogewsh** [comment]
> Your batch size is super small. How many params does your model have?

num. model params: 299239424 (num. trained: 299239424)
